server/models/Post.js

```diff
@@ -0,0 +1,72 @@
+const { init } = require("../dbConfig");
+const { ObjectId } = require("mongodb");
+
+class Post {
+  constructor(data) {
+    this.id = data.id;
+    this.title = data.title;
+    this.username = data.username;
+    this.body = data.body;
+    this.resources = data.resources;
+  }
+
+  static get all() {
+    return new Promise(async (resolve, reject) => {
+      try {
+        const db = await init();
+        const postData = await db.collection("posts").find().toArray();
+        const posts = postData.map(
+          (p) =>
+            new Post({
+              ...p,
+              id: p._id,
+              title: p.title,
+              username: p.username,
+              body: p.body,
+              resources: p.resources
+            })
+        );
+        resolve(posts);
+      } catch (err) {
+        reject("Error retrieving post");
+      }
+    });
+  }
+
+  static findById(id) {
+    return new Promise(async (resolve, reject) => {
+      try {
+        const db = await init();
+        let postData = await db
+          .collection("posts")
+          .find({ _id: ObjectId(id) })
+          .toArray();
+        let post = new Post({ ...postData[0], id: postData[0]._id });
+        resolve(post);
+      } catch (err) {
+        reject("Post not found");
```

Comment:
this error never gets thrown it just resolves with undefined if the post isn't found
this could be fixed by adding an if statement `if(!postData.length)` that throws an error

Reply:
thanks